fdbrpc/TLSConnection.actor.cpp

```diff
@@ -354,21 +447,21 @@ void TLSOptions::init_plugin() {
 		throw tls_error();
 	}
 
-	policyVerifyPeersSet = Reference<ITLSPolicy>( plugin->create_policy() );
-	if ( !policyVerifyPeersSet) {
+	policyVerifyPeersSet = AsyncVar<Reference<ITLSPolicy>>(Reference<ITLSPolicy>(plugin->create_policy()));
+	if ( !policyVerifyPeersSet.get()) {
 		// Hopefully create_policy logged something with the log func
 		TraceEvent(SevError, "TLSConnectionCreatePolicyVerifyPeersSetError");
 		throw tls_error();
 	}
 
-	policyVerifyPeersNotSet = Reference<ITLSPolicy>(plugin->create_policy());
-	if (!policyVerifyPeersNotSet) {
+	policyVerifyPeersNotSet = AsyncVar<Reference<ITLSPolicy>>(Reference<ITLSPolicy>(plugin->create_policy()));
+	if (!policyVerifyPeersNotSet.get()) {
 		// Hopefully create_policy logged something with the log func
 		TraceEvent(SevError, "TLSConnectionCreatePolicyVerifyPeersNotSetError");
 		throw tls_error();
 	}
 }
 
 bool TLSOptions::enabled() {
-	return !!policyVerifyPeersSet && !!policyVerifyPeersNotSet;
+	return !!policyVerifyPeersSet.get() && !!policyVerifyPeersNotSet.get();
```

Comment:
While this code is unchanged, it's not super easy to read.  The !! is serving only to cast the Reference to a bool and get its non-negated value, and the reason that's necessary is that while Reference has a bool cast operator it has no other method that returns bool.  Maybe we should correct that and add a `bool valid() const` method to Reference, which operator bool can then use?

Reply:
Fine, don't be in love with the "covert to bool" operator.

I mean, it's also not even necessary here.  I could have just written `policyVerifyPeersSet && policyVerifyPeersNotSet`, but I figured the !! was more obvious as to its bool-ish-ness.

BUT FINE.  HAVE YOUR `valid()`.  I HOPE YOU ENJOY IT.